Test-PR for CI

JKRhb opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description

<!--
Put here the description of your contribution:
- describe which part(s) of RIOT is (are) involved
- if it's a bug fix, describe the bug that it solves and how it is solved
- you can also give more information to reviewers about how to test your changes
-->


### Testing procedure

<!--
Details steps to test your contribution:
- which test/example to compile for which board and is there a 'test' command
- how to know that it was not working/available in master
- the expected success test output
-->


### Issues/PRs references

<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

JKRhb committed:
Lay foundation for using base URI from CoAP packet

JKRhb committed:
Add assertion for non-null base pointer

JKRhb committed:
Change order of header file includes

JKRhb committed:
Revert signature change and save base URI in the thing

JKRhb committed:
Update TODO

JKRhb committed:
Fix quotation marks in makefile

JKRhb committed:
Add function to get the IP address of the highest level

JKRhb committed:
Fix check for link_local address

JKRhb committed:
Remove "example.org" as fallback address

JKRhb committed:
Fix incorrect assertion in TD

JKRhb committed:
FIx pointer problems with get_base_address

JKRhb committed:
Remove commented out wot_thing_addr variable

JKRhb committed:
Change 0 initialization of ula and local address ip structs to {0}

Citrullin committed:
ifed for ledcontrol

Citrullin committed:
Fix res in for loop

JKRhb committed:
Replace gnrc ipv6 router modules with default ipv6 module

JKRhb committed:
Add default language tag

JKRhb committed:
Update wot_config.c

JKRhb committed:
Fix generation of default_language string

JKRhb committed:
Update wot_config.c

JKRhb committed:
Fix generation of default_language string

JKRhb committed:
Remove walrus operator from script for better compatibility

JKRhb committed:
Make first attempt of integrating generation script into build process

@Citrullin This is probably not the right way to do it (as it also does not
completely work right now) but it is a start, I guess :)

Citrullin committed:
Fix issues in wot_config.c

Citrullin committed:
Add fixme

Citrullin committed:
Some serilization fixes

JKRhb committed:
Add possibility to add thing models from build process

JKRhb committed:
Remove generation script from the tests directory

JKRhb committed:
Pass used modules to wot generation script

JKRhb committed:
Move generation Makefile to "makefiles" folder

JKRhb committed:
Add thing_instance_information to makefile and script

JKRhb committed:
Move definition of json_files to app makefile

Citrullin committed:
Fix included null pointer

JKRhb committed:
Remove resolved TODOs from generaion script

JKRhb committed:
Use "no security" as default scheme when no security is defined

JKRhb committed:
Update wot_config.c

JKRhb committed:
Implement proper timezone offset in generation script

JKRhb committed:
Remove debug print statements

JKRhb committed:
Add missing "href" field to link struct

JKRhb committed:
Add functionality to include multiple thing models and merge them

JKRhb committed:
Update example

Citrullin committed:
Refactor makefiles

Citrullin committed:
Refactor makefiles

JKRhb committed:
Export WOT variables

JKRhb committed:
Use python3 for build process

JKRhb committed:
Update python script to include app directory

Citrullin committed:
Fix some issues

JKRhb committed:
Fix error regarding the generation of type fields

JKRhb committed:
Add new makefile rule for generation of wot config

"all" depends now on this rule

JKRhb committed:
Replace SRC with GENSRC

Citrullin committed:
Some fixes

Citrullin committed:
Fix config

JKRhb committed:
Remove unused code

JKRhb committed:
Add proper check for required affordances of thing models

JKRhb committed:
Add (hopefully) correct sorting of CoAP resources

JKRhb committed:
Add TODO for inserting link parameters

JKRhb committed:
Revert "Add (hopefully) correct sorting of CoAP resources"

This reverts commit 4488075b61c49f719e32133b15819ac2516012df.

JKRhb committed:
Remove unneeded functions from generation script

JKRhb committed:
Allow for hrefs with hypens

JKRhb committed:
Allow for partial implementation of thing models

JKRhb committed:
Use namespace "riot_os" for RIOT specific settings in instance JSON

JKRhb committed:
Prevent @context duplicates

JKRhb committed:
Let build process on JSON files, remove thing model merging

JKRhb committed:
Retrieve default @language from @context (if it is set)

JKRhb committed:
Fix rule order in Makefile.gensrc

JKRhb committed:
Remove trailing whitespace

Citrullin committed:
Refactoring wot_forms in coap

Citrullin committed:
Fix security in wot definition

Citrullin committed:
Fix missing character for type

JKRhb committed:
Update generation script

JKRhb committed:
Remove obsolete add_hef function from generation script

JKRhb committed:
Update example JSON

JKRhb committed:
Add determination of default CoAP methods

JKRhb committed:
Fix issue with missing key of affordances

JKRhb committed:
Remove extension struct

JKRhb committed:
Fix security definitions in generation script

JKRhb committed:
Refactor copying of "forms" affordance field

JKRhb committed:
Copy security from the instance information

JKRhb committed:
Autoformat generation script

JKRhb committed:
Fix link parameters of TD resource

See https://github.com/w3c/wot-discovery/issues/120

JKRhb committed:
Add brackets around IPv6 base address

JKRhb committed:
Remove "zero_struct" CStruct option from python script

JKRhb committed:
Fix typos in python script

JKRhb committed:
Fix python style issues

JKRhb committed:
Fix whitespace error in coap_config/Makefile.gensrc

JKRhb committed:
Remove trailing whitespace

JKRhb committed:
Fix more typos

JKRhb committed:
Add C++ compatible headers

JKRhb committed:
Fix header guard formats

JKRhb committed:
fixup! Fix header guard formats

JKRhb committed:
Fix variable scopes

JKRhb committed:
Fix whitespace and newline issues

JKRhb committed:
Fix header guards

JKRhb committed:
Fix unneeded variable reassignment

JKRhb committed:
Remove APPLICATION from test Makefiles

JKRhb committed:
Add ARRAY_SIZE to old wot_coap test

JKRhb committed:
Fix problem with corrupted base addresses

JKRhb committed:
Fix previous_prop check for titles and descriptions

JKRhb committed:
Remove FIXME comment from timezone_offset

JKRhb committed:
Fix data_schema issues in WoT script

JKRhb committed:
Remove incorrectly set commas from TD

The issue occured after "type" fields.

JKRhb committed:
Fix serialization of min and max values

Serialize values without quotation marks.

JKRhb committed:
Remove quotation marks around brackets in TD

JKRhb committed:
Add commas between enum elements

JKRhb committed:
Rework TM system and add extension mechanism

JKRhb committed:
Move TD/TM definitions to its own module

JKRhb committed:
Refactor affordance names

JKRhb committed:
Remove obsolete code

JKRhb committed:
Add copyright and license information

JKRhb committed:
Refactor get statements

None is already the default so setting it
explicitly is unnecessary

JKRhb committed:
Add missing imports

JKRhb committed:
Fix type hint for CStruct children

JKRhb committed:
Fix signature of _generate_field method

This change removes a warning in the
ThingStruct class that inherits from the CStruct
class

JKRhb committed:
Clean up call of parser script

Don't pass BOARD and USEMODULE to the generator
script anymore, perform the call of the script
quietly

JKRhb committed:
Echo an info message when calling the script

JKRhb committed:
Make placeholders and bindings optional

JKRhb committed:
Fix variable name

JKRhb committed:
Change name of main function

Change the name of the main function according
to RIOTs coding convention for python scripts.

JKRhb committed:
Skip the WoT TD context in the generation script

JKRhb committed:
Add assertions for security definitions

JKRhb committed:
Change info text for no defined security

JKRhb committed:
Make both bindings and meta data optional

Allow for the user to not define any bindings or
meta data as long as the provided Thing Model
can be parsed by the generator script and contains
the necessary bindings and handler information.

JKRhb committed:
Rework data structures for generator script

Still WIP

JKRhb committed:
Fix key of "no security" scheme

JKRhb committed:
Fix assertion of version field

JKRhb committed:
Reformat code

JKRhb committed:
fixup! Rework data structures for generator script

Citrullin committed:
Reduce usage of [] and {}

Citrullin committed:
Refactor wot implementation

Citrullin committed:
Change fill_json in seerialization.c

JKRhb committed:
sys/net/coap: add WoT TD content format

JKRhb committed:
sys/net/wot: use WoT content format for TD

JKRhb committed:
Add missing function declarations

JKRhb committed:
Add missing header guards

JKRhb committed:
fixup! Add missing header guards

JKRhb committed:
fixup! Add missing header guards

JKRhb committed:
Make header files C++ compatible

JKRhb committed:
Fix header guard format

JKRhb committed:
Remove trailing whitespace

JKRhb committed:
fixup! Make header files C++ compatible

JKRhb committed:
fixup! Fix header guard format

JKRhb committed:
fixup! Fix header guard format

JKRhb committed:
Add header guards to security header files

JKRhb committed:
Fix issues caused by rebasing

JKRhb committed:
Clean up examples

JKRhb committed:
Remove outdated module path

JKRhb committed:
Remove obsolete WoT tests

JKRhb committed:
Add deleted esp dependency installation scripts

JKRhb committed:
Fix key of operation type field

JKRhb committed:
Remove obsolete listener search function

JKRhb committed:
Add DEBUG message

JKRhb committed:
Add DEBUG messages

JKRhb committed:
Fix bug in serialization of operation types

JKRhb committed:
Remove unused CoAP functions

JKRhb committed:
Disable debug statements

JKRhb committed:
fixup! Fix bug in serialization of operation types

JKRhb committed:
Add basic documentation to all python script classes

JKRhb committed:
Reformat python code

JKRhb committed:
Remove excessive newlines

JKRhb committed:
fixup! Some updates for WoT

JKRhb committed:
Add copyright headers

JKRhb committed:
fixup! Add copyright headers

JKRhb committed:
fixup! Add copyright headers

JKRhb committed:
fixup! Add copyright headers

JKRhb committed:
Adding missing newlines at file endings

JKRhb committed:
Address style issues

JKRhb committed:
Address style issues

JKRhb committed:
Address style issues

JKRhb committed:
Address style issues

JKRhb committed:
Address style issues

JKRhb committed:
Address style issues

JKRhb committed:
Address style issues

JKRhb committed:
Remove empty line

JKRhb committed:
Fix python license headers

JKRhb committed:
Remove obsolete config data

JKRhb committed:
Address issues in shell scripts

JKRhb commented on sys/include/net/wot/serialization/json_keys.h:
```diff
@@ -0,0 +1,78 @@
+const char _json_key_in[] = "in";
+const char _json_key_name[] = "name";
+const char _json_key_qop[] = "qop";
+// const char _json_key_scheme[] = "scheme";
```
@Citrullin I had to comment this line out due to a "multiple definitions" error – would it make sense to move the declarations in this file to a separate `.c` file?

Citrullin replied:
It probably makes sense to just remove it in json_writer.c. 
Guess I just didn't finished this part.

JKRhb committed:
Add documentation to io.h

JKRhb committed:
fiuxp! Add documentation to io.h

JKRhb committed:
Add documentation to serialization.h

JKRhb committed:
fixup! Add documentation to serialization.h

JKRhb committed:
Add documentation to persistance.h

JKRhb committed:
Add documentation to wot.h

JKRhb committed:
fixup! Add documentation to persistance.h

JKRhb committed:
fixup! Add documentation to wot.h

JKRhb committed:
fixup! Add documentation to wot.h

JKRhb committed:
fixup! Add documentation to wot.h

JKRhb committed:
Add documentation to coap.h

JKRhb committed:
Add documentation to config.h

JKRhb committed:
Add documentation to wot/config.h

JKRhb committed:
Add documentation to json_writer.h

JKRhb committed:
fixup! Add documentation to json_writer.h

JKRhb committed:
fixup! Add documentation to serialization.h

JKRhb committed:
fixup! Add documentation to wot.h

JKRhb committed:
fixup! Add documentation to wot.h

JKRhb committed:
fixup! Add documentation to wot.h

JKRhb committed:
Only regenerate instance fields for the first block

JKRhb committed:
fixup! Add documentation to wot.h

JKRhb committed:
Remove TODO regarding WoT Discovery

JKRhb committed:
Reformat arithmetic operations

JKRhb committed:
Reformat control statements

JKRhb committed:
fixup! Add documentation to coap.h

JKRhb committed:
Use coaps as base URL scheme if gcoap_dtls is loaded

JKRhb committed:
Add DTLS config to WoT CoAP Example

JKRhb committed:
Change MAX_ADRESSES_TO_CHECK to lowercase

JKRhb committed:
Include PSK security in TDs by default if gcoap_dtls is loaded

JKRhb committed:
Add use of DTLS credentials to WoT CoAP module

Co-Authored-By: schulztr <75017039+schulztr@users.noreply.github.com>

JKRhb committed:
Add DTLS credentials to WoT CoAP example

Co-Authored-By: schulztr <75017039+schulztr@users.noreply.github.com>

JKRhb committed:
Fix DTLS configuration in example

schulztr committed:
dist/tools/wot: make @type in TD optional

schulztr committed:
Support raw TDs

schulztr committed:
Fix default namespace

schulztr committed:
Fix NumberSchema

schulztr committed:
Fix Description and Events

JKRhb committed:
sys/net/wot/coap: fix bug in base regeneration

The fact that the opening IPv6 address bracket
was not null-terminated led to an error during
serialization.

JKRhb committed:
sys/net/wot/coap: use strcpy for "[" of base address

JKRhb committed:
Extract PSK identity from CFLAGS

JKRhb committed:
Fix typo in struct generation script

JKRhb committed:
Add NULL check for security schemes

schulztr committed:
fix coding convention

schulztr committed:
fix coding convention

Citrullin committed:
Remove unused functions

schulztr committed:
Apply coding style suggestions from code review

Co-authored-by: Jan Romann <jan.romann@uni-bremen.de>

Citrullin committed:
Add enable/disable feature for WoT COAP

Citrullin committed:
Make title(s)/description(s) TD conform

Citrullin committed:
Add COAP_FORMAT_WOT_TD to excluded patterns

Citrullin committed:
Fix documentation and formatting

Citrullin committed:
Fix misspelling of Length

Citrullin committed:
Add documentation

Citrullin committed:
Some format fixes

schulztr committed:
Distinguish Thing Model by @type

schulztr committed:
Fixed small bug in Thing Model determination.

schulztr committed:
dist/tools/wot: support secured resources

schulztr committed:
sys/net/coap: change gcoap socket type

schulztr committed:
sys/net/wot/core/serialization/: fix bug in context serialization

schulztr committed:
sys/net/wot/core: include serialization subdirectory into core directory to avoid multiple compilation of the wot_core-module

schulztr committed:
dist/tools/wot: fix coding style

JKRhb committed:
fix: add missing add_prefix to Makefile.gensrc

JKRhb committed:
fixup! fix: add missing add_prefix to Makefile.gensrc

JKRhb committed:
fixup! fixup! fix: add missing add_prefix to Makefile.gensrc

schulztr committed:
dist/tools/wot: fix bug concerning empty resource list

schulztr committed:
tests/wot_coap_generation/config/wot_td: add Thing Model prefix type

schulztr committed:
Merge pull request #31 from namib-project/wot_gcoap_fix

Fix bug concerning empty resource list and add the Thing Model prefix in tests/wot_coap_generation

schulztr committed:
sys/net/wot/core: fix bug concerning security definition in forms

schulztr committed:
sys/net/wot/core: remove redundant function _wot_td_security_writer

schulztr committed:
Merge pull request #32 from namib-project/wot_gcoap_fix

sys/net/wot/core: fix bug concerning security definition in forms

JKRhb committed:
sys/net/wot/coap: use correct well-known URI

Instead of /.well-known/wot-thing-description, the Discovery
specification now uses /.well-known/wot instead.

JKRhb committed:
sys/net/nanocoap: simplify obtaining accept option

JKRhb committed:
sys/net/wot/coap: support content negotiation

JKRhb committed:
sys/coap: add Size2 option number

JKRhb committed:
sys/net/wot/coap: add size2 support
